src/viam/examples/modules/complex/summation/api.cpp

```diff
@@ -0,0 +1,126 @@
+#include <vector>
+
+#include <google/protobuf/descriptor.h>
+
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource.hpp>
+#include <viam/sdk/rpc/server.hpp>
+
+#include "../proto/summation.grpc.pb.h"
+#include "../proto/summation.pb.h"
+#include "api.hpp"
+
+using namespace viam::sdk;
+using namespace viam::service::summation::v1;
+
+/* SummationRegistration methods */
+
+SummationRegistration::SummationRegistration(
+    const google::protobuf::ServiceDescriptor* service_descriptor)
+    : ResourceRegistration(service_descriptor){};
+
+std::shared_ptr<ResourceServer> SummationRegistration::create_resource_server(
+    std::shared_ptr<ResourceManager> manager) {
+    return std::make_shared<SummationServer>(manager);
+};
+
+std::shared_ptr<Resource> SummationRegistration::create_rpc_client(
+    std::string name, std::shared_ptr<grpc::Channel> chan) {
+    return std::make_shared<SummationClient>(std::move(name), std::move(chan));
+};
+
+/* Summation methods */
+
+std::shared_ptr<ResourceRegistration> Summation::resource_registration() {
+    const google::protobuf::DescriptorPool* p = google::protobuf::DescriptorPool::generated_pool();
+    const google::protobuf::ServiceDescriptor* sd =
+        p->FindServiceByName(SummationService::service_full_name());
+    if (!sd) {
+        throw std::runtime_error("Unable to get service descriptor for the summation service");
+    }
+    return std::make_shared<SummationRegistration>(sd);
+}
+
+API Summation::static_api() {
+    return {"viam", "service", "summation"};
+}
+
+API Summation::dynamic_api() const {
+    return static_api();
+}
+
+Summation::Summation(std::string name) : Service(std::move(name)){};
+
+/* Summation server methods */
+
+SummationServer::SummationServer() : ResourceServer(std::make_shared<ResourceManager>()){};
+SummationServer::SummationServer(std::shared_ptr<ResourceManager> manager)
+    : ResourceServer(manager){};
+
+std::vector<double> repeated_field_to_vector(const google::protobuf::RepeatedField<double>& f) {
```

Comment:
an identical method exists in `camera.cpp`. instead of duplicating, can we move it to some shared space?

Reply:
Moved to `common/utils` along with `vector_to_repeated_field` and renamed both methods to indicate they only handle doubles.